pandas/tests/dtypes/cast/test_can_hold_element.py

```diff
@@ -53,3 +53,18 @@ def test_can_hold_element_int_values_float_ndarray():
     # integer but not losslessly castable to int64
     element = np.array([3, 2 ** 65], dtype=np.float64)
     assert not can_hold_element(arr, element)
+
+
+def test_can_hold_element_int8_int():
+    arr = np.array([], dtype=np.int8)
+
+    element = 2
+    assert can_hold_element(arr, element)
```

Comment:
can you add a case where the array *cannot* hold?

Reply:
a branch is in the works.  there are places where fixing can_hold_element breaks other things